manual/core/retries/README.md

```diff
@@ -0,0 +1,153 @@
+consistency level. The coordinator replied with an `UNAVAILABLE` error.
+
+If the policy rethrows the error, the user code will get an [UnavailableException]. You can inspect
+the exception's fields to get the amount of replicas that were known to be *alive* when the error
+was triggered, as well as the amount of replicas that where *required* by the requested consistency
+level.
+
+The default policy triggers a maximum of one retry, to the next node in the query plan. The
+rationale is that the first coordinator might have been network-isolated from all other nodes
+(thinking they're down), but still able to communicate with the client; in that case, retrying on
+the same node has almost no chance of success, but moving to the next node might solve the issue.
+
+### onReadTimeout
+
+A read request reached the coordinator, which initially believed that there were enough live
+replicas to process it. But one or several replicas were too slow to answer within the predefined
+timeout (`read_request_timeout_in_ms` in `cassandra.yaml`); therefore the coordinator replied to the
+client with a `READ_TIMEOUT` error.
+
+This could be due to a temporary overloading of these replicas, or even that they just failed or
+were turned off. During reads, Cassandra doesn't request data from every replica to minimize
+internal network traffic; instead, some replicas are only asked for a checksum of the data. A read
+timeout may occur even if enough replicas responded to fulfill the consistency level, but only
+checksum responses were received (the method's `dataPresent` parameter allow you to check if you're
+in that situation).
+
+If the policy rethrows the error, the user code will get a [ReadTimeoutException] (do not confuse
+this error with [DriverTimeoutException], which happens when the coordinator didn't reply at all to
+the client).
+
+The default policy triggers a maximum of one retry (to the same node), and only if enough replicas
+had responded to the read request but data was not retrieved amongst those. That usually means that
+enough replicas are alive to satisfy the consistency, but the coordinator picked a dead one for data
+retrieval, not having detected that replica as dead yet. The reasoning is that by the time we get
+the timeout, the dead replica will likely have been detected as dead and the retry has a high chance
+of success.
+
+### onWriteTimeout
+
+This is similar to `onReadTimeout`, but for write operations. The reason reads and writes are
+handled separately is because a read is obviously a non mutating operation, whereas a write is
+likely to be. If a write times out at the coordinator level, there is no way to know whether the
+mutation was applied or not on the non-answering replica.
+
+If the policy rethrows the error, the user code will get a [WriteTimeoutException].
+
+This method is only invoked for [idempotent](../idempotence/) statements. Otherwise, the driver
```

Comment:
We should also port the idempotence docs

Reply:
They're already in #972 because I also needed them there.

Reply:
Ah, didn't catch that, nice!